api.yml

```diff
@@ -185,66 +83,166 @@ paths:
           description: not supported by this implementation!
 components:
   schemas:
-    IssueRequest:
+    ConstructCredentialRequest:
       type: object
+      required:
+        - issuer
+        - vcTemplateReference
+        - claims
+        - output
       properties:
         issuer:
           type: string
-          description: Optional identifier for the issuer of the credential. If this is missing, it may be pre-configured or implied by other values in the request.
-          example: did:example:76e12ec712ebc6f1c221ebfeb1f
-        subject:
+          description: A reference to the issuer's pre-configured profile.
+        vcTemplateReference:
           type: string
-          description: Optional identifier for the subject of the credential. If this is missing, it may be pre-configured or implied by other values in the request, or there may not be an identifier for the subject of the credential.
-          example: did:example:ebfeb1f712ebc6f1c276e12ec21
-        types:
+          description: A reference to the pre-configured Verifiable Credential template.
+        claims:
           type: array
+          description: A set of claims about one or more subjects.
           items:
-            type: string
-          description: Optional value for the type(s) of a credential. If this is missing, it may be pre-configured or implied by other values in the request.
-          example: UniversityDegreeCredential
-        issuanceDate:
-          type: string
-          format: date-time
-          description: Optional value for the issuance date and time of the credential, to be used only when this value should not be automatically set to the current date and time.
-          example: '2010-01-01T19:23:24Z'
-        expirationDate:
+            $ref: '#/components/schemas/Claims'
+        output:
+          $ref: '#/components/schemas/Output'
+      example: {
+        "issuer": "did:example:123456677",
+        "vcTemplateReference": "https://issuer.oidp.uscis.gov/schemas/v1/prcard",
+        "claims": [{
+          "sub": "did:example:123456abcdefg",
```

Comment:
Why is this `sub`? The Verifiable Credentials Data Model notes that this should be `id`... what am I missing?

Reply:
yes... this blending of JWT and VC data model terms is thing we also encountered when updating the well known did configuration. https://identity.foundation/.well-known/did-configuration.json

https://github.com/decentralized-identity/.well-known

IMO this another valueable part of this API, we can help clarify how to handle these cases... in well known did config, we opted to keep this as close the vc data model as possible, but we did use some of the jwt reserved fields...

Reply:
@msporny

> Why is this `sub`? The Verifiable Credentials Data Model notes that this should be `id`... what am I missing?

sub == id

claims == credentialSubject

We can discuss names but I am more interested in allowing multiple credential subjects in the VC.

Reply:
One of the motivations for the current API was that it should not be biased toward JSON-LD, JWT, or any other of the existing credential formats.